Automattic/wp-calypso
# Plugins: Add upload reducer

Part of #15665. No visual differences.

Add reducers for plugin upload state. Similar to existing [theme upload reducers](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/upload-theme/reducer.js).

Also tweaks the actions to:
* Store calypso plugin id instead of jetpack plugin id (`slug` from jetpack)
* Simplify progress by storing only percentage uploaded

**To Test**
* Run unit tests, or dispatch actions from js console and check state in `state.plugins.upload`
